stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -500,34 +481,42 @@ internal class StripePaymentController internal constructor(
         )
     }
 
+    private suspend fun dispatchSource(
+        source: Source,
+        callback: ApiResultCallback<Source>
+    ) = withContext(uiContext) {
+        logger.debug("Dispatching Source with id=${source.id}")
+        callback.onSuccess(source)
+    }
+
     private suspend fun dispatchPaymentIntentResult(
         paymentIntentResult: PaymentIntentResult,
         callback: ApiResultCallback<PaymentIntentResult>
-    ) = withContext(Dispatchers.Main) {
+    ) = withContext(uiContext) {
         logger.debug("Dispatching PaymentIntentResult for ${paymentIntentResult.intent.id}")
         callback.onSuccess(paymentIntentResult)
     }
 
     private suspend fun dispatchSetupIntentResult(
         setupIntentResult: SetupIntentResult,
         callback: ApiResultCallback<SetupIntentResult>
-    ) = withContext(Dispatchers.Main) {
+    ) = withContext(uiContext) {
         logger.debug("Dispatching SetupIntentResult for ${setupIntentResult.intent.id}")
         callback.onSuccess(setupIntentResult)
     }
```

Comment:
can we templatize them?

```
private suspend inline fun <reified ResultModel: StripeModel> dispatchResultOnUiContext(
        model: ResultModel,
        id: String,
        callback: ApiResultCallback<ResultModel>
    ) = withContext(uiContext) {
        logger.debug("Dispatching ${ResultModel::class.java.simpleName} for $id")
        callback.onSuccess(model)
    }
```

Reply:
`dispatchSetupIntentResult` is no longer being used